src/libs/actions/Session/index.js

```diff
@@ -50,21 +50,18 @@ function setSuccessfulSignInData(data) {
  * @param {String} login
  */
 function createAccount(login) {
-    Onyx.merge(ONYXKEYS.SESSION, {error: ''});
+    Onyx.merge(ONYXKEYS.ACCOUNT, {error: ''});
 
     API.User_SignUp({
         email: login,
     }).then((response) => {
-        if (response.jsonCode === 200) {
+        // A 405 means that the account needs to be validated. We should let the user proceed to the ResendValidationForm view.
+        if (response.jsonCode === 200 || response.jsonCode === 405) {
             return;
         }
 
-        let errorMessage = response.message || `Unknown API Error: ${response.jsonCode}`;
-        if (!response.message && response.jsonCode === 405) {
-            errorMessage = 'Cannot create an account that is under a controlled domain';
-        }
-        Onyx.merge(ONYXKEYS.SESSION, {error: errorMessage});
         Onyx.merge(ONYXKEYS.CREDENTIALS, {login: null});
+        Onyx.merge(ONYXKEYS.ACCOUNT, {error: response.message || `Unknown API Error: ${response.jsonCode}`});
```

Comment:
Just to add some context to this change... When we set the login to `null` we get redirected back to the `LoginForm`? And we are handling `account` not `session` errors there?

Reply:
Correct!